runtime/vm/ContinuationHelpers.cpp

```diff
@@ -995,6 +995,7 @@ preparePinnedVirtualThreadForUnmount(J9VMThread *currentThread, j9object_t syncO
 										| OBJECT_HEADER_LOCK_FLC;
 						}
 						j9objectmonitor_t const oldValue = lock;
+						VM_AtomicSupport::writeBarrier();
```

Comment:
A write barrier is generally not required before the CAS. What is the purpose of this write barrier?

Reply:
This is to match the usage in monhelpers.c line 97. I suppose the barrier will act as an guard that other writes to lockword will not overwrite what is being CAS'ed.